services/source/updater.c

```diff
@@ -0,0 +1,355 @@
+        addr_info.addr = new_address;
+        priv_eeprom_set_app_info(addr_info);
+        break;
+    }
+    case 'G':
+    {
+        if (new_address < GOLD_MINIMUM_ADDR) {
+            snprintf(pcWriteBuffer, xWriteBufferLen, "Address too low for golden image\n");
+            return pdFALSE;
+        }
+        image_info addr_info = eeprom_get_golden_info();
+        addr_info.addr = new_address;
+        priv_eeprom_set_golden_info(addr_info);
+        break;
+    }
+    default:
+        snprintf(pcWriteBuffer, xWriteBufferLen, "Invalid Image Type\n");
+        return pdFALSE;
+    }
+    snprintf(pcWriteBuffer, xWriteBufferLen, "Address updated\n");
+    return pdFALSE;
+}
+
+static BaseType_t prvInfoCommand(char *pcWriteBuffer, size_t xWriteBufferLen, const char *pcCommandString) {
+    int32_t image_param_len = 0;
+    const char *image = FreeRTOS_CLIGetParameter(
+                    /* The command string itself. */
+                    pcCommandString,
+                    /* Return the next parameter. */
+                    1,
+                    /* Store the parameter string length. */
+                    &image_param_len);
+    if (image_param_len > 1) {
+        snprintf(pcWriteBuffer, xWriteBufferLen, "Invalid Image Type\n");
+        return pdFALSE;
+    }
+
+    image_info info;
+    switch (*image) {
+    case 'A':
+        info = eeprom_get_app_info();
+        break;
+    case 'G':
+        info = eeprom_get_golden_info();
+        break;
+    default:
+        snprintf(pcWriteBuffer, xWriteBufferLen, "Invalid Image Type\n");
+        return pdFALSE;
+    }
+    snprintf(pcWriteBuffer, xWriteBufferLen, "Exists: %d\nSize: %d\nAddr: 0x%08X\nCrc: %04X\n",
+             info.exists,
+             info.size,
+             info.addr,
+             info.crc);
+    return pdFALSE;
+
+}
+
+static BaseType_t prvErrorCommand(char *pcWriteBuffer, size_t xWriteBufferLen, const char *pcCommandString) {
+    snprintf(pcWriteBuffer, xWriteBufferLen, "Error: %s\n", error);
+    error = "None";
+    return pdFALSE;
+}
+
+static const CLI_Command_Definition_t xAddrCommand = {"address", "address:\n\tSet address of image. Image is either G or A\n\tFirst parameter is image type, second parameter is address as hex\n", prvAddrCommand, 2};
+static const CLI_Command_Definition_t xInfoCommand = {"info", "info:\n\tGet Image info. Image is either G or A\n", prvInfoCommand, 1};
+static const CLI_Command_Definition_t xErrorCommand = {"error", "error:\n\tGet latest error in updater module", prvErrorCommand, 0};
+
+/**
+ * @brief
+ *      Attempts to malloc a buffer
+ * @detail
+ *      This function will try to malloc smaller and smaller buffers until it succeeds
+ *      In this case, having any buffer at all is more important than size
+ * @param buf
+ *      Pointer to the pointer to malloc the buffer in to
+ * @return
+ *      size of buffer that was allocated for
+ */
+void *get_buffer(int32_t size) {
+    return pvPortMalloc(size);
```

Comment:
The function doesn't really seem to be doing what the comments say it is doing

Reply:
no, it doesn't. Whoops